src/policy/fees.cpp

```diff
@@ -1010,7 +1010,8 @@ FeeFilterRounder::FeeFilterRounder(const CFeeRate& minIncrementalFee)
 CAmount FeeFilterRounder::round(CAmount currentMinFee)
 {
     std::set<double>::iterator it = feeset.lower_bound(currentMinFee);
-    if ((it != feeset.begin() && insecure_rand.rand32() % 3 != 0) || it == feeset.end()) {
+    const auto rand = WITH_LOCK(m_insecure_rand_mutex, return insecure_rand.rand32());
```

Comment:
adfd07564d5544f327d7432faddcf9612a6a22b3

Just noting that now now `rand32()` is always called.

Reply:
Could have 

```c++
    bool PreferLower() { LOCK(m_insecure_rand_mutex); return insecure_rand.rand32() % 3 == 0; }

    if (it != m_fee_set.begin() && PreferLower() || it == m_fee_set.end()) --it;
```

Reply:
Done, thanks!